src/cls/rgw/cls_rgw_client.cc

```diff
@@ -209,12 +209,13 @@ void cls_rgw_bucket_complete_op(ObjectWriteOperation& o, RGWModifyOp op, string&
   o.exec(RGW_CLASS, RGW_BUCKET_COMPLETE_OP, in);
 }
 
-static bool issue_bucket_list_op(librados::IoCtx& io_ctx, const string& oid,
-				 const cls_rgw_obj_key& start_obj,
-				 const string& filter_prefix,
-				 uint32_t num_entries, bool list_versions,
-				 BucketIndexAioManager *manager,
-				 rgw_cls_list_ret *pdata) {
+void cls_rgw_bucket_list(librados::ObjectReadOperation& op,
```

Comment:
this just sets up the op, yes?

Reply:
that's probably a lexical pattern, but maybe in future we could make that visible in the name.

Reply:
how about `cls_rgw_bucket_list_op()`?

Reply:
that's a good ctor, yeah